crates/autopilot/src/database/onchain_order_events/ethflow_events.rs

```diff
@@ -69,12 +96,22 @@ impl OnchainOrderParsing<EthFlowData, EthOrderPlacement> for EthFlowOnchainOrder
         order: &Order,
         hashmap: &HashMap<EventIndex, EthFlowData>,
         _onchain_order_placement: &OnchainOrderPlacement,
-    ) -> EthOrderPlacement {
-        EthOrderPlacement {
-            uid: order.uid,
-            // unwrap is allowed, as any missing event_index would have been filtered beforehand
-            // by the implementation of the function parse_custom_event_data
-            valid_to: hashmap.get(event_index).unwrap().user_valid_to as i64,
+    ) -> EthFlowDataForDb {
+        EthFlowDataForDb {
+            eth_order_placement: EthOrderPlacement {
+                uid: order.uid,
+                // unwrap is allowed, as any missing event_index would have been filtered beforehand
+                // by the implementation of the function parse_custom_event_data
+                valid_to: hashmap.get(event_index).unwrap().user_valid_to as i64,
+            },
+            pre_interaction: Interaction {
+                // For ethflow orders, the owner is always the ethflow contract
+                target: ByteArray(order.owner.0),
+                value: BigDecimal::new(0.into(), 1),
+                // 4c84c1c8 is the identifier of the following function:
+                // https://github.com/cowprotocol/ethflowcontract/blob/main/src/CoWSwapEthFlow.sol#L57
+                data: hex!("4c84c1c8").to_vec(),
```

Comment:
ubernit: Can this be a constant?

```rust
const WRAP_ALL_SELECTOR: [u8; 4] = hex!("...");
```

And then:
```suggestion
                data: WRAP_ALL_SELECTOR.to_vec(),
```

This way, its more immediately apparent what the interaction does (i.e. calls the `wrapAll` method). Might also make sense to repurpose the comment to explain why the interaction is needed instead of what the selector hex means.